src/clients/Client.cc

```diff
@@ -455,33 +455,34 @@ sameUrlHosts(const char *url1, const char *url2)
 static void
 purgeEntriesByHeader(HttpRequest *req, const char *reqUrl, Http::Message *rep, Http::HdrType hdr)
 {
-    const char *hdrUrl, *absUrl;
-
-    absUrl = NULL;
-    hdrUrl = rep->header.getStr(hdr);
-    if (hdrUrl == NULL) {
+    const auto hdrUrl = rep->header.getStr(hdr);
+    if (!hdrUrl)
         return;
-    }
 
     /*
      * If the URL is relative, make it absolute so we can find it.
      * If it's absolute, make sure the host parts match to avoid DOS attacks
      * as per RFC 2616 13.10.
      */
+    SBuf absUrlMaker;
+    const char *absUrl;
```

Comment:
Nit: I recommend explicitly initializing this variable (for general style/safety reasons) even though it is not technically necessary. I do not insist on this change.

```suggestion
    const char *absUrl = nullptr;
```

P.S. Sorry I forgot to include this nit in the updated review I just submitted.

Reply:
Done in 83b191fdb